openwisp-config/files/openwisp.agent

```diff
@@ -460,8 +464,11 @@ test_configuration() {
 }
 
 perform_default_test() {
-	# max 3 attempts to get checksum
-	for i in $(seq 1 3); do
+
+	sleep $TEST_DELAY
```

Comment:
look here: https://github.com/openwisp/openwisp-config/blob/92d209c4995e64e455513a0d2336224f6e751439/openwisp-config/files/openwisp.agent#L328-L354

There's already a `$sleep_time` variable that is based on an argument of the `apply_configuration` function and defaults to 5 seconds, right?

In turn, before `perform_default_test` is called, `apply_configuration` is called with no arguments, which will use the default 5 seconds sleep time, right?

So it should be possible to make that existing sleep time configurable instead of adding a new one, this would have the advantage that the `post_reload_hook` would be called after the right delay.

If we think about it, the reason you need TEST_DELAY is because you're waiting for the config to be applied, so it makes sense to convert this config option into the option which allows us to control how much to wait after the config is reloaded before proceeding with anything else. Something like `POST_RELOAD_DELAY`.
If we go this route, we can then also use the same delay here: https://github.com/openwisp/openwisp-config/blob/92d209c4995e64e455513a0d2336224f6e751439/openwisp-config/files/openwisp.agent#L580-L590.

What do you think?

Reply:
Yes, this seems like the way to go. So we basically need to change the following:

* rename `TEST_DELAY` to `POST_RELOAD_DELAY`
* remove the `sleep` in `perform_default_test`
* make the `sleep` in `apply_configuration` and in `restore_backup` use `POST_RELOAD_DELAY`

@nemesisdesign Do you agree?

Reply:
@okraits yes, perfect!

Reply:
@nemesisdesign I updated the PR accordingly.